packages/rest/src/lib/REST.ts

```diff
@@ -434,16 +434,20 @@ export class REST extends AsyncEventEmitter<RestEventsMap> {
 			};
 		}
 
-		const majorIdMatch = /^\/(?:channels|guilds|webhooks)\/(\d{17,19})/.exec(endpoint);
+		const majorIdMatch = /(?:^\/webhooks\/(\d{17,19}\/[^/]+))|(?:^\/(?:channels|guilds|webhooks)\/(\d{17,19}))/.exec(
+			endpoint,
+		);
 
-		// Get the major id for this route - global otherwise
-		const majorId = majorIdMatch?.[1] ?? 'global';
+		// Get the major id or id + token for this route - global otherwise
+		const majorId = majorIdMatch?.[1] ?? majorIdMatch?.[2] ?? 'global';
```

Comment:
```suggestion
		const majorId = majorIdMatch?.[2] ?? majorIdMatch?.[1] ?? 'global';
```

Flipping these because the standard match will happen far more often and when matching a webhook it stops processing after the first match